### ISSUE TITLE
ApplicationSet Name can no longer be more than 63 chars

### ISSUE THREAD

**Issue by dyatlov** [argoproj/argo-cd#15282]
This is a regression from 2.7.
This commit introduced it: https://github.com/argoproj/argo-cd/commit/7192eabaf4715f73eeb2e132df95d3a15762cd09

A seemingly information-only label makes it impossible to have applicationsets with longer names because from now on that name is copied into the tracking label.

We'd appreciate if that was configurable to be able to turn it on / off or override etc.

**Comment by ericblackburn** [comment]
Let's please get this prioritized.  It is a complete road block.

**Comment by ericblackburn** [comment]
Also, i agree with your [comment here](https://github.com/argoproj/argo-cd/pull/13456#issuecomment-1698100331) 

I noted on the argo cli issue that they should just grab the info from the k8s appset object.  The information is already stored there.

Given the original goal was to query for appsets via the argo cli, #14285, I think #13456 should either be removed or changed to annotations, the correct way to store information. 

I really thing the Argo product needs to avoid using labels at all cost to link argo resources based their names.  Argo is meant to be a k8s deployment tool. It should prioritize good hygine and not introduce bad practices that can spread when others see code like this and copy pasta the idea.  So, just having an On/Off switch would make Argo fragile and anti-k8s.

**Comment by todaywasawesome** [comment]
Recommendation from @rbreeze is that we move to annotation tracking as @ericblackburn has suggested. Anyone want to make a PR?

**Comment by speedfl** [comment]
Annotation tracking is also the recommended solution for app in any namespace.
https://argo-cd.readthedocs.io/en/stable/operator-manual/app-any-namespace/#switch-resource-tracking-method

I can start working on a PR today :+1:

**Comment by rumstead** [comment]
What is the plan for enabling the cli to query for the child applications with the annotation?

**Comment by ericblackburn** [comment]
The argo cli query might need to be a separate initiative.  We should fix the critical blocking bug first and design a nice to have cli query solution later.

**Comment by speedfl** [comment]
@rumstead we can think about it later but here are some solution

- Using jsonPath

```
argocd app list --filter='$.metadata.ownerReferences[?(@.kind=='ApplicationSet')].name=test-appset'
argocd app list --filter='$.metadata.annotations.['argocd.argoproj.io/appset-instance']=test-appset'
```

> jq style could also be used (argocd already uses gojq)

- Using a new field

```
argocd app list --appset-name=test-appset
```

> an example is `--app-namespace` that was introduced for application in any namespace feature

- Using a field-selector...

**Comment by dyatlov** [comment]
@speedfl does that imply that we have to load a full application list into client before filtering it? I would rather filter applications on server side before returning (might help with a further application pagination initiative)

**Comment by speedfl** [comment]
You are right @dyatlov above proposals imply loading all applications on client side. 

I was just pointing that we could find other solutions compared to the existing one which brings the 63 char regression

**Comment by rumstead** [comment]
> argocd app list --filter='$.metadata.ownerReferences[?(@.kind=='ApplicationSet')].name=test-appset'

We wouldn't even need to add the annotation in this case :)

**Comment by rumstead** [comment]
based on the PR we can probably close this now?

**Comment by jannfis** [comment]
A fix for this has been released in v2.8.4
